src/MatrixTypes_PETSc.f90

```diff
@@ -311,7 +320,11 @@ SUBROUTINE transpose_PETScMatrixType(matrix)
   !This is to avoid a deadlock in IBarrier in MPICH
   CALL PetscCommBuildTwoSidedSetType(matrix%comm, &
       PETSC_BUILDTWOSIDED_ALLREDUCE,iperr)
+#if ((PETSC_VERSION_MAJOR>=3) && (PETSC_VERSION_MINOR>6))
+  CALL MatTranspose(matrix%a,MAT_INPLACE_MATRIX,matrix%a,iperr)
+#else
   CALL MatTranspose(matrix%a,MAT_REUSE_MATRIX,matrix%a,iperr)
+#endif
```

Comment:
Well that's an unnecessary name change...

Reply:
yeah...not a huge fan